sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -126,71 +135,77 @@ export class AssessmentSequentialProcessingService {
   }
 
   /**
-   * Finds all applications before the one related to the {@link assessmentId} provided
-   * and returns the sum of all awards associated with non-overwritten applications.
+   * Get the program year awards totals for part-time and
+   * full-time and contribution totals for full-time for the assessment.
+   * @param assessmentId assessment id.
+   * @param options method options.
+   * - `offeringIntensity` the offering intensity to be used.
+   * - `alternativeReferenceDate` the reference date to be used.
+   * @returns the promise to get the program year totals.
+   */
+  async getProgramYearTotals(
+    assessmentId: number,
+    options?: {
+      offeringIntensity?: OfferingIntensity;
+      alternativeReferenceDate?: Date;
+    },
+  ): Promise<ProgramYearTotal> {
+    // Get the current assessment from the assessment id.
+    const currentAssessment = await this.getCurrentAssessment(assessmentId);
+    // The chronology of the applications is defined by the method {@link getSequencedApplications}.
+    // Only the current assessment awards are considered since it must reflect the most updated
+    // workflow calculated values.
+    const sequencedApplications = await this.getSequencedApplications(
+      currentAssessment.application.applicationNumber,
+      currentAssessment.application.student.id,
+      currentAssessment.application.programYear.id,
+      { alternativeReferenceDate: options?.alternativeReferenceDate },
+    );
+    // Get the application numbers of the previous applications.
+    const applicationNumbers = sequencedApplications.previous.map(
+      (application) => application.applicationNumber,
+    );
+
+    const [programYearTotalAwards, ftProgramYearContributionTotal] =
+      await Promise.all([
+        // Get the program year awards totals for part-time and full-time.
+        await this.getProgramYearPreviousAwardsTotals(
+          sequencedApplications,
+          currentAssessment,
+          options,
+        ),
+        // Only get the full-time contribution totals if the offering intensity is full-time.
+        OfferingIntensity.fullTime === options.offeringIntensity &&
+        applicationNumbers?.length > 0
+          ? await this.getProgramYearContributionTotals(applicationNumbers)
+          : null,
+      ]);
+    return {
+      awardTotal: programYearTotalAwards,
+      ftProgramYearContributionTotal: ftProgramYearContributionTotal,
+    };
+  }
```

Comment:
As suggestion, it can be adjusted as below.
```ts
// Only get the full-time contribution totals if the offering intensity is full-time.
const shouldGetProgramYearContributionTotals =
  OfferingIntensity.fullTime === options.offeringIntensity &&
  !!applicationNumbers?.length;
const [awardTotals, contributionTotals] = await Promise.all([
  // Get the program year awards totals for part-time and full-time.
  await this.getProgramYearPreviousAwardsTotals(
    sequencedApplications,
    currentAssessment,
    options,
  ),
  shouldGetProgramYearContributionTotals
    ? await this.getProgramYearContributionTotals(applicationNumbers)
    : null,
]);
return {
  awardTotals,
  contributionTotals,
};
```

Reply:
I am taking this, it looks cleaner